app/javascript/app/document-capture/components/full-screen.jsx

```diff
@@ -0,0 +1,54 @@
+import React, { useRef, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import createFocusTrap from 'focus-trap';
+import Image from './image';
+import useI18n from '../hooks/use-i18n';
+
+function FullScreen({ onRequestClose, children }) {
+  const t = useI18n();
+  const modalRef = useRef(/** @type {?HTMLDivElement} */ (null));
+  const trapRef = useRef(/** @type {?import('focus-trap').FocusTrap} */ (null));
```

Comment:
React newb here (again)... are there more react-friendly libraries than the `focus-trap` one that we could use? Seems like a bit of a kludge to use it here?

Reply:
> React newb here (again)... are there more react-friendly libraries than the `focus-trap` one that we could use? Seems like a bit of a kludge to use it here?

The `focus-trap` library itself [has a React integration](https://github.com/focus-trap/focus-trap-react), though a combination of the maintenance warning and [use of deprecated features](https://github.com/focus-trap/focus-trap-react/blob/749f02180e8c7cb6c237723419fbb81e739c261a/src/focus-trap-react.js#L49) ([documentation](https://reactjs.org/docs/react-dom.html#finddomnode)) gives me a bit of pause.

That being said, we could definitely follow a similar pattern and extract something a bit more abstracted and expressive. I could imagine either a hook or a component like the one in `focus-trap-react`, where both encapsulate the look logic here.

Something like:

```jsx
function useFocusTrap() {
  const trapRef = useRef(/** @type {?import('focus-trap').FocusTrap} */ (null));
  const onRequestCloseRef = useRef(onRequestClose);
  useEffect(() => {
    // Since the focus trap is only initialized once, but the callback could
    // be changed, ensure that the current reference is kept as a mutable value
    // to reference in the deactivation.
    onRequestCloseRef.current = onRequestClose;
  }, [onRequestClose]);
  useEffect(() => {
    trapRef.current = createFocusTrap(modalRef.current, {
      onDeactivate: () => onRequestCloseRef.current(),
    });
    trapRef.current.activate();
    return trapRef.current.deactivate;
  }, []);
  return trapRef;
}

// ...

function FullScreen() {
  const trapRef = useFocusTrap();

  return <div ref={ trapRef } { /* ... */ } />;
}
```

Reply:
> are there more react-friendly libraries than the `focus-trap` one

To the broader question, if the intention is to pull in more of USWDS / Login Design System, it's worth noting that the design system implements its own focus trap:

https://github.com/uswds/uswds/blob/50f6ffd/src/js/utils/focus-trap.js

It would be possible — albeit clunky, like in #3957 — to use the logic directly, which would have a small advantage in ensuring a consistent behavior, and eventually having a small reduction of the overall JavaScript bundle size.

But the implementation is rather basic, and doesn't support things like closing upon <kbd>Escape</kbd> presses. And bringing it back to this discussion, the <kbd>Escape</kbd> press handling is the primary thing which makes the `focus-trap` integration a kludge 😅